### ISSUE TITLE
Input mutating original options array of objects with __mask_ value and __original key

### ISSUE THREAD

**Issue by rahman-anz** [formkit/formkit#1437]
### Reproduction

https://formkit.link/7d15a183c77989975f508535643229ef

### Describe the bug

When appending values to an already used options array, the original array is mutated to __mask_ value and __original key. Saw the same behavior in dropdown, autocomplete and radio inputs. And if the original array is non-mutable like a computed or a prop, we get a warning in the console like the following

![image (2)](https://github.com/formkit/formkit/assets/130835929/cbd95a6d-d257-485b-9e6e-3bf7ce73b03c)

Going through the closed issues, saw this one which is very similar. 
https://github.com/formkit/formkit/issues/1046

But the issue is still unresolved in Formkit version 1.6.5 and pro version 0.126.12 

**Update**:  On closer inspection, saw that we don't even need to append values on top of the original array. The original array is getting mutated nevertheless.

https://formkit.link/3cbbc5da22bf257b4bcfb6aee9b8079f

### Environment

• OS: Windows 11
• Browser Chrome
• Version 1.6.5

**Comment by justin-schroeder** [comment]
This is true and intentional. Using frozen or non writable objects will cause problems in this case. While we could potentiallly do this differently this is how it works.

As a work around you should not have this mutation if your values are strings.
